modin/experimental/core/execution/native/implementations/hdk_on_native/dataframe/dataframe.py

```diff
@@ -1887,7 +1887,7 @@ def filter(self, key):
         )
 
         if self is base:
-            exprs = OrderedDict()
+            exprs = dict()
```

Comment:
What is the problem with OrderedDict?

Reply:
To avoid mixing a regular dict (which is returned by one of the functions) with this type.